### PR TITLE
Add version formatting for NGINX pod label

### PR DESCRIPTION
**Pull request by jjngx**
### Proposed changes

This PR adds a method for formatting NGINX version.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by pdabelf5 — COMMENTED** [review]
Would it make sense to have the version functionality in its own file?

**Review by shaun-nx — APPROVED** [review]

File: internal/nginx/version.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package nginx
```
Comment (vepatel) [inline internal/nginx/version.go:1]: seems like few lines here aren't covered by version_test.go, Is it possible to add few negatives cases?

File: internal/nginx/version.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	return nv
+}
+
+// String returns the raw Nginx version string from `nginx -v`.
+func (v Version) String() string {
+	return v.Raw
+}
+
+// Format returns a string representing Nginx version.
+func (v Version) Format() string {
```
Comment (pdabelf5) [inline internal/nginx/version.go:56]: This is to be called in `main.go` ?

Comment (jjngx) [reply internal/nginx/version.go:56]: Yes, will be dressed in a separate PR

> This is to be called in `main.go` ?

Yes, will be dressed in a separate PR

**Review by pdabelf5 — APPROVED** [review]

**Pull request merged**